src/hooks/useDrag.tsx

```diff
@@ -406,6 +545,45 @@ export const onDelete = (
         newHighlightErrors
       )
     }
+    case "announcement": {
+      const announcementKeyExist = !_.isEmpty(
+        frontMatter.sections.find((section) =>
+          EditorHomepageFrontmatterSection.isAnnouncements(section)
+        )
+      )
+      if (!announcementKeyExist) {
+        // should not reach here, but defensively return the original state
+        return homepageState
+      }
+
+      const announcementsIndex = frontMatter.sections.findIndex((section) =>
+        EditorHomepageFrontmatterSection.isAnnouncements(section)
+      )
+      const announcementsSection: AnnouncementsFrontmatterSection = frontMatter
+        .sections[announcementsIndex] as AnnouncementsFrontmatterSection
+
+      const newAnnouncementOptions = deleteElement(
+        announcementsSection.announcements.announcement_items,
+        indexToDelete
+      )
+      const newAnnouncementErrors = deleteElement(
+        errors.announcementItems,
+        indexToDelete
+      )
+
+      const newDisplayAnnouncements = deleteElement(
+        displayAnnouncementItems,
+        indexToDelete
+      )
```

Comment:
nit: some amount of repetition here as it's very similar to the above. not a big issue for now as intent is clear but we might benefit it this model extends also to other section types

Reply:
deprio for now ya